### DISCUSSION TITLE
Can interopRequireDefault be customized?

### DISCUSSION THREAD

**Discussion by schacker** [swc-project/swc#2568] [Q&A] [answered]
# Describe
The document describes it like this   `https://swc.rs/docs/config-js-module#strict `
> To prevent the __esModule property from being exported, you can set the strict option to true.

## input
```ts
import path from 'path'
import Hobber from '@ones/hobber-core'

const appPath = path.resolve(__dirname)
const app = Hobber({ appPath: appPath })
app.then((hobber)=>{ hobber.lift() });

export default {}
```
# Config
build script: "swc-build": "npx swc src -d ./dist --only **/*.ts --config-file .swcrc --env-name='test' --copy-files"
"@swc/cli": "^0.1.51",
"@swc/core": "^1.2.103"
```json
{
  "minify": false,
  "sourceMaps": true,
  "module": {
    "type": "commonjs",
    "strict": false, // key config
    "strictMode": true,
    "lazy": false,
    "noInterop": false,
    "ignoreDynamic": false
  }
}
```

But, my output looks like this:
```js
"use strict";
Object.defineProperty(exports, "a", {
    value: !0
}), exports.default = void 0;
var _path = _interopRequireDefault(require("path"));
var _hobberCore = _interopRequireDefault(require("@ones/hobber-core")); // has __esModule build with  babel or tsc
function _interopRequireDefault(a) {
    return a && a.a ? a : {
        default: a
    };
}
var appPath = _path.default.resolve(__dirname);
_hobberCore.default({
    b: appPath
}).then(function(a) {
    a.c();
}), exports.default = {
};
```
# if set strict to true, like this:
```js
"use strict";
exports.default = void 0;
var _path = _interopRequireDefault(require("path"));
var _hobberCore = _interopRequireDefault(require("@ones/hobber-core"));
function _interopRequireDefault(a) {
    return a && a.a ? a : {
        default: a
    };
}
var appPath = _path.default.resolve(__dirname);
_hobberCore.default({
    b: appPath
}).then(function(a) {
    a.c();
}), exports.default = {
};
```

# Why `a` not `__esModule`?

**Comment by kdy1** [comment]
Because `__esModule` is not standard.

**Reply by schacker** [reply]
> Because `__esModule` is not standard.

so, how to solve this incompatible usage?

**Reply by kdy1** [reply]
Can you try `import * as Hobber from '@ones/hobber-core'`?

**Reply by kdy1** [reply]
Or `import { default as Hobber } from '@ones/hobber-core'`

**Reply by schacker** [reply]
> Or `import { default as Hobber } from '@ones/hobber-core'`

It doesn't work either.

**Comment by schacker** [comment]
How to be compatible with babel ?
if __esModule is not supported, many libraries compiled by babel or tsc cannot be used.

**Reply by kdy1** [reply]
Use `strict: false`

**Reply by schacker** [reply]
not working~  
The key here is that babel and tsc are compiled into `__esModule`, not `a`  

The two are not compatible!

**Reply by kdy1** [reply]
You are wrong about `__esModule`. It's just a boolean flag, not a module.

**Comment by schacker** [comment, marked as answer]
key config: jsc.minify.mangle.property

When I delete the property, the result is correct!
